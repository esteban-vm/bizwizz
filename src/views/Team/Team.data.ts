import type { TeamMateProps } from './TeamMate'

const teamMates: TeamMateProps[] = [
  { id: 'fec6d982-1994-4d02-9cc7-18a86719fb7c', name: 'Paula Lee', designation: 'CEO' },
  { id: 'c512f1ff-6fff-4aee-be97-7c0697204186', name: 'Ted Johnson', designation: 'Co-Founder' },
  { id: '37d5c8a6-3234-42f1-8379-65b50f5f12cb', name: 'Miranda Shaffer', designation: 'Lead Designer' },
]

export default teamMates
