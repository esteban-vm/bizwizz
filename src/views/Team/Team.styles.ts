import clsx from 'clsx'

const styles = {
  wrapper: clsx('flex flex-wrap justify-evenly'),
}

export default styles
