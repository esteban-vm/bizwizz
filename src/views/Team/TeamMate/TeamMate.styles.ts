import clsx from 'clsx'

const styles = {
  wrapper: clsx(
    'relative h-96 w-96 select-none rounded-full border-[3px] border-solid border-primary bg-cover bg-center',
    "first:bg-[url('/images/team-mate-1.jpg')] last:bg-[url('/images/team-mate-3.jpg')] even:bg-[url('/images/team-mate-2.jpg')]",
    '[&>div:last-child]:hover:top-1/2 [&>div]:hover:opacity-100'
  ),
  overlay: clsx(
    'absolute left-0 top-0 h-full w-full rounded-full bg-primary/70',
    'opacity-0 transition-opacity duration-200'
  ),
  content: clsx(
    'absolute left-1/2 top-[30%] -translate-x-1/2 -translate-y-1/2 text-white opacity-0 transition-all duration-500'
  ),
  name: clsx('text-[2.7rem] font-semibold'),
  designation: clsx('text-2xl font-medium'),
  links: clsx('mt-8 flex justify-between'),
  icon: clsx('m-[2.6px] h-12 w-12 fill-white hover:fill-primary-light'),
}

export default styles
