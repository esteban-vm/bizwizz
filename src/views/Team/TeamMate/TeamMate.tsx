import type { FC, ID, IconName } from '@/types'
import { Icon } from '@/components'

export type TeamMateProps = {
  id: ID
  name: Capitalize<string>
  designation: string
}

const socialMediaLinks: IconName[] = ['facebook', 'twitter', 'linkedin']

const TeamMate: FC<TeamMateProps> = ({ id, name, designation }) => {
  return (
    <article id={id} className='team_mate'>
      <div className='team_mate_overlay' />
      <div className='team_mate_content'>
        <h4 className='team_mate_name'>{name}</h4>
        <p className='team_mate_designation'>{designation}</p>
        <div className='team_mate_links'>
          {socialMediaLinks.map((link, index) => (
            <a key={index} href={`https://www.${link}.com`} target='_blank' title={link}>
              <Icon name={link} className='team_mate_icon' />
            </a>
          ))}
        </div>
      </div>
    </article>
  )
}

export default TeamMate
