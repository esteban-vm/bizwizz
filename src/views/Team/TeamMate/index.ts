export { default, type TeamMateProps } from './TeamMate'
