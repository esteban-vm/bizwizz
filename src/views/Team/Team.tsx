import type { FC } from '@/types'
import { Wave } from '@/components'
import TeamMate, { type TeamMateProps } from './TeamMate'
import styles from './Team.styles'

export const teamMates: TeamMateProps[] = [
  { id: 'fec6d982-1994-4d02-9cc7-18a86719fb7c', name: 'Paula Lee', designation: 'CEO' },
  { id: 'c512f1ff-6fff-4aee-be97-7c0697204186', name: 'Ted Johnson', designation: 'Co-Founder' },
  { id: '37d5c8a6-3234-42f1-8379-65b50f5f12cb', name: 'Miranda Shaffer', designation: 'Lead Designer' },
]

const Team: FC = () => {
  return (
    <section aria-labelledby='team_title' className='section bg-black' id='team'>
      <Wave is='white' />
      <div className='section_header'>
        <h2 className='section_heading' id='team_title'>
          Meet our team
        </h2>
        <h3 className='section_subheading'>Lorem ipsum dolor sit amet</h3>
      </div>
      <div className={styles.team_mates}>
        {teamMates.map((teamMate) => (
          <TeamMate key={teamMate.id} {...teamMate} />
        ))}
      </div>
    </section>
  )
}

export default Team
