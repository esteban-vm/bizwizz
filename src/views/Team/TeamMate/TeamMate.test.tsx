import { render, cleanup, within, screen } from '@/tests'
import { teamMates } from '../Team'
import TeamMate from './TeamMate'

describe('Team Mate:', () => {
  let asFragment: () => DocumentFragment
  const [testTeamMate] = teamMates

  beforeEach(() => {
    void ({ asFragment } = render(<TeamMate {...testTeamMate} />))
  })

  afterEach(cleanup)

  it('should render correctly', () => {
    expect(asFragment()).toMatchSnapshot()
  })

  it('should have correct structure', () => {
    const teamMateElem = screen.getByRole('article')
    expect(teamMateElem).toBeInTheDocument()
    expect(teamMateElem).toHaveAttribute('id', testTeamMate.id)
    expect(within(teamMateElem).getByRole('heading', { name: testTeamMate.name, level: 4 })).toBeInTheDocument()
    expect(within(teamMateElem).getByText(testTeamMate.designation)).toBeInTheDocument()
    expect(within(teamMateElem).getByRole('link', { description: /facebook/i })).toBeInTheDocument()
    expect(within(teamMateElem).getByRole('link', { description: /twitter/i })).toBeInTheDocument()
    expect(within(teamMateElem).getByRole('link', { description: /linkedin/i })).toBeInTheDocument()
  })
})
