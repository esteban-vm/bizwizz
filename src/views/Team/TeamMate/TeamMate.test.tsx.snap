// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Team Mate: > should render correctly 1`] = `
<DocumentFragment>
  <article
    class="team_mate"
    id="fec6d982-1994-4d02-9cc7-18a86719fb7c"
  >
    <div
      class="team_mate_overlay"
    />
    <div
      class="team_mate_content"
    >
      <h4
        class="team_mate_name"
      >
        Paula Lee
      </h4>
      <p
        class="team_mate_designation"
      >
        CEO
      </p>
      <div
        class="team_mate_links"
      >
        <a
          href="https://www.facebook.com"
          target="_blank"
          title="facebook"
        >
          <svg
            class="team_mate_icon"
            role="img"
          >
            <use
              xlink:href="#icon-facebook"
            />
          </svg>
        </a>
        <a
          href="https://www.twitter.com"
          target="_blank"
          title="twitter"
        >
          <svg
            class="team_mate_icon"
            role="img"
          >
            <use
              xlink:href="#icon-twitter"
            />
          </svg>
        </a>
        <a
          href="https://www.linkedin.com"
          target="_blank"
          title="linkedin"
        >
          <svg
            class="team_mate_icon"
            role="img"
          >
            <use
              xlink:href="#icon-linkedin"
            />
          </svg>
        </a>
      </div>
    </div>
  </article>
</DocumentFragment>
`;
